compiler/noirc_frontend/src/hir_def/types/arithmetic.rs

```diff
@@ -71,11 +71,13 @@ impl Type {
                 let evaluate = |typ: &Type| {
                     typ.evaluate_to_signed_field_helper(&kind, dummy_location, run_simplifications)
                 };
+                let lhs_evaluated = evaluate(lhs);
+                let rhs_evaluated = evaluate(rhs);
```

Comment:
I think the evaluations were done "on the fly" to avoid having to evaluate them when not needed. So this will make things slower, though it's not clear if it's noticeable (but I'd still prefer to do things faster if possible).